chia/util/chia_logging.py

```diff
@@ -38,56 +40,59 @@ def get_file_log_handler(
             colorlog.ColoredFormatter(
                 f"%(asctime)s.%(msecs)03d {service_name} %(name)-{file_name_length}s: "
                 f"%(log_color)s%(levelname)-8s%(reset)s %(message)s",
                 datefmt=log_date_format,
                 reset=True,
             )
         )
-
-        logger = colorlog.getLogger()
-        logger.addHandler(handler)
+        handlers.append(stdout_handler)
     else:
-        logger = logging.getLogger()
-        logger.addHandler(get_file_log_handler(file_log_formatter, root_path, logging_config))
+        handlers.append(get_file_log_handler(file_log_formatter, root_path, logging_config))
 
     if logging_config.get("log_syslog", False):
         log_syslog_host = logging_config.get("log_syslog_host", "localhost")
         log_syslog_port = logging_config.get("log_syslog_port", 514)
         log_syslog_handler = SysLogHandler(address=(log_syslog_host, log_syslog_port))
         log_syslog_handler.setFormatter(logging.Formatter(fmt=f"{service_name} %(message)s", datefmt=log_date_format))
-        logger = logging.getLogger()
-        logger.addHandler(log_syslog_handler)
-
-    if "log_level" in logging_config:
-        if logging_config["log_level"] == "CRITICAL":
-            logger.setLevel(logging.CRITICAL)
-        elif logging_config["log_level"] == "ERROR":
-            logger.setLevel(logging.ERROR)
-        elif logging_config["log_level"] == "WARNING":
-            logger.setLevel(logging.WARNING)
-        elif logging_config["log_level"] == "INFO":
-            logger.setLevel(logging.INFO)
-        elif logging_config["log_level"] == "DEBUG":
-            logger.setLevel(logging.DEBUG)
-            logging.getLogger("aiosqlite").setLevel(logging.INFO)  # Too much logging on debug level
-        else:
-            logger.setLevel(logging.INFO)
-    else:
-        logger.setLevel(logging.INFO)
+        handlers.append(log_syslog_handler)
 
     if beta_root_path is not None:
-        logger.addHandler(get_file_log_handler(file_log_formatter, beta_root_path, get_beta_logging_config()))
+        handlers.append(get_file_log_handler(file_log_formatter, beta_root_path, get_beta_logging_config()))
+
+    root_logger = logging.getLogger()
+    log_level_invalid = False
+    for handler in handlers:
+        try:
+            handler.setLevel(log_level)
+        except Exception:
+            handler.setLevel(default_log_level)
+            log_level_invalid = True
+        root_logger.addHandler(handler)
+
+    if log_level_invalid:
+        root_logger.error(
+            f"Invalid log level found in {service_name} config: {log_level}. Defaulting to: {default_log_level}"
+        )
+
+    # Adjust the root logger to the smallest used log level since its default level is WARNING which would overwrite
+    # the potentially smaller log levels of specific handlers.
+    root_logger.setLevel(min(handler.level for handler in handlers))
+
+    if root_logger.level < logging.INFO:
```

Comment:
I think we want this.  The log level could be in between info and debug without triggering the aiosqlite debug so the boundary for this should be where debug actually triggers.

```suggestion
    if root_logger.level <= logging.DEBUG:
```

Reply:
Ops, actually could have just accepted here but see https://github.com/Chia-Network/chia-blockchain/pull/13569/commits/2e54d82d2a0e077ce6633b444fb50ef91aa1bada